contracts/Comptroller/Diamond/facets/MarketFacet.sol

```diff
@@ -112,6 +130,53 @@ contract MarketFacet is IMarketFacet, FacetBase {
         return results;
     }
 
+    /**
+     * @notice Unlist an market by setting isListed to false
+     * @dev Pauses all actions, sets borrow/supply caps to 0 and sets collateral factor to 0.
+     * @param market The address of the market (token) to unlist
+     * @return uint256 0=success, otherwise a failure. (See enum Error for details)
+     */
+    function unlistMarket(address market) external returns (uint256) {
+        ensureAllowed("unlistMarket(address)");
+
+        Market storage _market = markets[market];
+
+        if (!_market.isListed) {
+            return fail(Error.MARKET_NOT_LISTED, FailureInfo.UNLIST_MARKET_NOT_LISTED);
+        }
+
+        _actionPaused[market][uint256(Action.BORROW)] = true;
+        _actionPaused[market][uint256(Action.MINT)] = true;
+        _actionPaused[market][uint256(Action.REDEEM)] = true;
+        _actionPaused[market][uint256(Action.REPAY)] = true;
+        _actionPaused[market][uint256(Action.ENTER_MARKET)] = true;
+        _actionPaused[market][uint256(Action.LIQUIDATE)] = true;
```

Comment:
I think we could delegate on the other admin actions to perform these changes, and in this function only require the state we want. So, I wouldn't:
* pause the actions
* set borrow and supply caps to zero
* set CF to zero

I would *require* instead:
* every action is paused
* borrow and supply caps are zero
* CF is zero

This way, to unlist a market in a VIP, we could add the following commands:
* pause every action
* set borrow and supply cap to zero
* set CF to zero
* invoke unlistMarket

This would allow `unlistMarket` to be smaller, and easier to test and maintain. At the end, if the "requires" are successful, this function will only set to false the `isListed` attribute and emit the `MarketUnlisted` event.

Reply:
Another consequence of this: we wouldn't move to `FacetBase` some of the events we currently have in `SetterFacet`